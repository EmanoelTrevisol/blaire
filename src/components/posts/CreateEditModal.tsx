import React from 'react';
import { View, Modal, StyleSheet, Text, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import { Post } from '@models/Post';
import PostForm from './PostForm';

export interface IProps {
  onSubmit: ({ title, body }: { title: string; body: string }) => Promise<void>;
  isSubmitting: boolean;
  close: () => void;
  title: string;
  buttonTitle: string;
  visible: boolean;
  post: Post | null;
  isCreate?: boolean;
}

const CreateEditModal = (props: IProps) => {
  const {
    onSubmit,
    close,
    title,
    buttonTitle,
    visible,
    post,
    isSubmitting,
    isCreate,
  } = props;

  return (
    <Modal visible={visible} transparent={true} animationType="slide">
      <View style={stl.modal}>
        <View style={stl.modalHeader}>
          <Text style={stl.modalTitle}>{title}</Text>
          <TouchableOpacity style={stl.modalCloseButton} onPress={close}>
            <Icon style={stl.modalCloseButtonIcon} name="times" />
          </TouchableOpacity>
        </View>
        <View style={stl.modalBody}>
          <PostForm
            isCreate={isCreate}
            onSubmit={onSubmit}
            isSubmitting={isSubmitting}
            buttonTitle={buttonTitle}
            post={post}
          />
        </View>
      </View>
    </Modal>
  );
};

const stl = StyleSheet.create({
  modal: {
    marginTop: '30%',
    padding: 10,
    backgroundColor: '#fff',
    flex: 1,
    borderTopEndRadius: 20,
    borderTopLeftRadius: 20,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.3,
    shadowRadius: 4,
    elevation: 5,
  },
  modalBody: {
    // flex: 1,
  },
  modalHeader: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    position: 'relative',
  },
  modalTitle: {
    fontSize: 24,
    fontWeight: 'bold',
  },
  modalCloseButton: {
    position: 'absolute',
    right: 0,
    top: 0,
  },
  modalCloseButtonIcon: {
    fontSize: 20,
    right: 0,
    marginRight: 0,
  },
});

export default CreateEditModal;
