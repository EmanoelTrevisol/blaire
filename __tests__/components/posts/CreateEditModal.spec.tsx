import React from 'react';
import CreateEditModal, { IProps } from '@/components/posts/CreateEditModal';
import { render } from '@testing-library/react-native';

let props: IProps;

describe('PostCard.tsx', () => {
  const renderComp = (cpProps: IProps) => {
    return render(<CreateEditModal {...cpProps} />);
  };

  beforeEach(() => {
    props = {
      title: 'Novo post',
      buttonTitle: 'Criar',
      post: null,
      onSubmit: jest.fn(),
      close: jest.fn(),
      visible: true,
    };
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  test('matches snapshot', () => {
    const { toJSON } = renderComp(props);

    expect(toJSON()).toMatchSnapshot();
  });
});
