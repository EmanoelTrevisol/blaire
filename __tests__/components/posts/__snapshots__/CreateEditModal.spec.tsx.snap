// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard.tsx matches snapshot 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#fff",
            "borderRadius": 20,
            "elevation": 5,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "padding": 20,
            "paddingBottom": 30,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Novo post
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 20,
                    "marginRight": 0,
                    "right": 0,
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={Object {}}
        >
          <View
            style={
              Object {
                "marginBottom": 60,
              }
            }
          >
            <View
              style={
                Object {
                  "marginVertical": 20,
                  "maxHeight": 80,
                }
              }
            >
              <View
                style={
                  Object {
                    "marginVertical": 5,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                >
                  Título
                </Text>
              </View>
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                editable={true}
                keyboardType="email-address"
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="Crie seu título"
                placeholderTextColor="#898989"
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderBottomColor": "#000",
                    "borderBottomWidth": 1,
                    "paddingVertical": 5,
                  }
                }
                textContentType="username"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Object {
                  "marginVertical": 20,
                  "maxHeight": 80,
                }
              }
            >
              <View
                style={
                  Object {
                    "marginVertical": 5,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                >
                  Post
                </Text>
              </View>
              <TextInput
                allowFontScaling={true}
                editable={true}
                multiline={true}
                numberOfLines={5}
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="Mal posso esperar para ler o que você tem para contar"
                placeholderTextColor="#898989"
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderBottomColor": "#000",
                    "borderBottomWidth": 1,
                    "paddingVertical": 5,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
          <View
            style={Object {}}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#ea94a5",
                  "borderRadius": 50,
                  "height": 50,
                  "opacity": 1,
                  "padding": 16,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }
              >
                Criar
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
